wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CampaignContributionTotalCron.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @file
+ */
+
+use Civi\Api4\Campaign;
+use Civi\Api4\Contribution;
+
+/**
+ * API specification (optional)
+ *
+ * @param array $spec
+ */
+function _civicrm_api3_goonjcustom_campaign_contribution_total_cron_spec(&$spec) {
+}
+
+/**
+ * Custom API to calculate and update total contribution amount per campaign.
+ *
+ * @param array $params
+ *
+ * @return array
+ *
+ * @throws \CRM_Core_Exception
+ */
+function civicrm_api3_goonjcustom_campaign_contribution_total_cron($params) {
```

Comment:
Can we follow Single Responsibility Principle here?
Right now the this single function handles data retrieval, calculation, updates, and message formatting